src/server/rdb_load.cc

```diff
@@ -2627,10 +2627,8 @@ void RdbLoader::PerformPreLoad(Service* service) {
   if (cmd == nullptr)
     return;  // MacOS
 
-  shard_set->AwaitRunningOnShardQueue([](EngineShard* es) {
-    for (const auto& name : es->search_indices()->GetIndexNames())
-      es->search_indices()->DropIndex(name);
-  });
+  shard_set->AwaitRunningOnShardQueue(
+      [](EngineShard* es) { es->search_indices()->DropAllIndices(); });
```

Comment:
And now please validate than FlushAll which is called before replication and loading a dataset cleans the indices. The initial goal of moving the drop into FlushDb was to get rid of this code entirely 🙂